addon/components/document-validity.js

```diff
@@ -0,0 +1,40 @@
+import { getOwner } from "@ember/application";
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { dropTask } from "ember-concurrency-decorators";
+
+import { parseDocument } from "ember-caluma/lib/parsers";
+
+/**
+ * Component to check the validity of a document
+ *
+ * ```hbs
+ * <DocumentValidity @document={{this.calumaDocument}} as |isValid|>
+ *  {{#if isValid}}
+ *    <p>The document is valid</p>
+ *  {{/if}}
+ * </DocumentValidity>
+ * ```
+ *
+ * @class DocumentValidity
+ * @yield {Boolean} isValid
+ */
+export default class DocumentValidity extends Component {
+  /**
+   * The document to be validated
+   * @argument {Document} document
+   */
+
+  @tracked isValid = null;
+
+  @dropTask
+  *validate() {
+    const document = getOwner(this)
+      .factoryFor("caluma-model:document")
+      .create({ raw: parseDocument(this.args.document) });
```

Comment:
According to the API docs, the user passes this down to the component. I'd welcome that.

Reply:
Wouldn't it be easier if the document generation was handled by caluma?

Reply:
You are using the cf-content component which already does that for you. Generating documents is really expensive in terms of performance.

Reply:
How would the host app access the document from cf-content then?

Reply:
Changed back to use document

Reply:
Ah ok, document is being yielded in content.

Reply:
Yes, exactly. Or you'd create a custom widget using this component which gets the document as well.